src/UI/Layout.hs

```diff
@@ -115,16 +135,39 @@ instance Show a => Show1 (LayoutF a) where
     Inset by child -> showsBinaryWith showsPrec sp "Inset" d by child
     Offset by child -> showsBinaryWith showsPrec sp "Offset" d by child
     GetMaxSize -> showString "GetMaxSize"
+    Align alignment child -> showsBinaryWith showsPrec sp "AlignLeft" d alignment child
 
 instance Eq2 LayoutF where
   liftEq2 eqA eqF l1 l2 = case (l1, l2) of
     (Inset s1 c1, Inset s2 c2) -> liftEq eqA s1 s2 && eqF c1 c2
     (Offset p1 c1, Offset p2 c2) -> liftEq eqA p1 p2 && eqF c1 c2
     (GetMaxSize, GetMaxSize) -> True
+    (Align a1 c1, Align a2 c2) -> a1 == a2 && eqF c1 c2
     _ -> False
 
 instance Eq a => Eq1 (LayoutF a) where
   liftEq = liftEq2 (==)
 
 instance (Eq a, Eq f) => Eq (LayoutF a f) where
   (==) = liftEq (==)
+
+instance Listable2 LayoutF where
+  liftTiers2 t1 t2
+    =  liftCons2 (liftTiers t1) t2 Inset
+    \/ liftCons2 (liftTiers t1) t2 Offset
+    \/ liftCons2 tiers t2 Align
+
+instance Listable a => Listable1 (LayoutF a) where
+  liftTiers = liftTiers2 tiers
+
+instance (Typeable a, Typeable b, Listable a, Listable b) => Listable (LayoutF a b) where
+  tiers = case eqT :: Maybe (b :~: Size (Maybe a)) of
+    Just Refl -> tiers1 \/ cons0 GetMaxSize
+    Nothing   -> tiers1
```

Comment:
We can only produce `GetMaxSize` for some instantiations of `b`. The `Refl` pattern match enables this.